src/main/java/com/jfrog/ide/idea/configuration/XrayServerConfigImpl.java

```diff
@@ -197,10 +199,30 @@ void setPassword(String password) {
     }
 
     void setCredentials(Credentials credentials) {
+        if (credentials == null) {
+            return;
+        }
         setUsername(credentials.getUserName());
         setPassword(credentials.getPasswordAsString());
     }
 
+    /**
+     * Initialize connection details from environment variables.
+     * Minimal required details are url and username.
+     */
+    public void initConnectionDetailsFromEnv() {
+        String urlEnv = System.getenv(URL_ENV);
+        String usernameEnv = System.getenv(USERNAME_ENV);
+        String passwordEnv = System.getenv(PASSWORD_ENV);
+        if (StringUtils.isBlank(urlEnv) || StringUtils.isBlank(usernameEnv)) {
+            return;
+        }
+        setUrl(urlEnv);
+        setUsername(usernameEnv);
+        setPassword(passwordEnv);
+        addCredentialsToPasswordSafe();
```

Comment:
I'm not sure that it is necessary in this case. We won't want to bother the user with password prompts when it is not really necessary.
```suggestion
```

Reply:
We decided to read the config-details once, and store them for the user.
Then, in future uses if the configuration is already set, we are not reading env-vars again.